README.md
![alt tag](https://raw.githubusercontent.com/MatterOverdrive/MatterOverdrive/1.12.2/MatterOverdriveLogo.png)

[![Jenkins](https://img.shields.io/jenkins/s/http/jenkins.k-4u.nl/job/MatterOverdrive/.svg?style=for-the-badge)](http://jenkins.k-4u.nl/view/Astro/job/CLib/)
[![GitHub issues](https://img.shields.io/github/issues-raw/MatterOverdrive/MatterOverdrive.svg?style=for-the-badge)](https://github.com/MatterOverdrive/MatterOverdrive/issues)
[![](https://img.shields.io/github/issues-pr-raw/MatterOverdrive/MatterOverdrive.svg?style=for-the-badge)](https://github.com/MatterOverdrive/MatterOverdrive/pulls)
[![license](https://img.shields.io/github/license/matteroverdrive/matteroverdrive.svg?style=for-the-badge)](https://github.com/MatterOverdrive/MatterOverdrive/blob/1.12.2/LICENSE.md)
[![GitHub code size in bytes](https://img.shields.io/github/languages/code-size/MatterOverdrive/MatterOverdrive.svg?style=for-the-badge)](https://github.com/MatterOverdrive/MatterOverdrive)
[![GitHub contributors](https://img.shields.io/github/contributors/MatterOverdrive/MatterOverdrive.svg?style=for-the-badge)](https://github.com/MatterOverdrive/MatterOverdrive/graphs/contributors)
[![Discord](https://img.shields.io/discord/364844705864744961.svg?style=for-the-badge)](https://discord.gg/erEyCsW)

##### Current Developer: TheCodedOne

## Table of Contents
* [About](#about)
* [Features](#features)
* [Contacts](#contacts)
* [Issues](#issues)
* [Building](#building)
* [IMC](#imc)

## About
Matter Overdrive is a Minecraft mod inspired by the popular Sci-fi TV series Star Trek. It dwells in the concept of replicating and transforming one type matter into another.
Although it may seem overpowered, Matter Overdrive takes a more realistic approach and requires the player to build a complex system before even the simplest replication can be achieved.

## Contacts
* [@TheCodedOne](https://twitter.com/TheCodedOne)
* [Dev_Osmium](mailto:devosmium@gmail.com)
* [@DevOsmium](https://twitter.com/DevOsmium)
* [Website](http://devosmium.ddns.net/matteroverdrive/matteroverdrive.html)
* [Discord](https://discord.gg/PC5GXyQ)

## Features
* [Matter Scanner](https://mo.simeonradivoev.com/items/matter_scanner/), for scanning matter patterns for replication.
* [Replicator](https://mo.simeonradivoev.com/items/replicator/), for transforming materials.
* [Decomposer](https://mo.simeonradivoev.com/items/decomposer/), for breaking down materials to basic form.
* [Transporter](https://mo.simeonradivoev.com/items/transporter/), for beaming up.
* [Phaser](https://mo.simeonradivoev.com/items/phaser/), to set on stun.
* [Fusion Reactors](https://mo.simeonradivoev.com/fusion-reactor/) and [Gravitational Anomaly](https://mo.simeonradivoev.com/items/gravitational_anomaly/)
* Complex Networking for replication control.
* Star Maps, with Galaxies, Stars and Planets
* [Androids](https://mo.simeonradivoev.com/android-guide/), become an Android and learn powerful RPG like abilities, such as Teleportation and Forefield Shields.


![Matter Overdrive Blocks and Items](https://media-elerium.cursecdn.com/attachments/210/237/main_screenshot.png)

## Issues
If you have any crashes, problems or suggestions just open a [new Issue](https://github.com/MatterOverdrive/MatterOverdrive/issues/new).
If your crash or problem was fixed, but is not yet released as a public download you can always download the latest [Dev Build](https://mo.simeonradivoev.com/builds/).

## Building
1. Clone this repository via 
  - SSH `git clone git@github.com:MatterOverdrive/MatterOverdrive.git` or 
  - HTTPS `git clone https://github.com/MatterOverdrive/MatterOverdrive.git`
2. Setup workspace (This currently does not work due to a ping timeout on rx14's mvn server.)
  - Decompiled source `gradlew setupDecompWorkspace`
  - Obfuscated source `gradlew setupDevWorkspace`
  - CI server `gradlew setupCIWorkspace`
3. Build `gradlew build`. Jar will be in `build/libs`
4. For core developer: Setup IDE
  - IntelliJ: Import into IDE and execute `gradlew genIntellijRuns` afterwards
  - Eclipse: execute `gradlew eclipse`
  - Don't forget to install [Lombok](https://projectlombok.org)
  
## IMC
See the example on [IMC](https://github.com/MatterOverdrive/MatterOverdrive/blob/1.12.2/src/main/java/matteroverdrive/api/IMC.java) or you can see the [IMC handler](https://github.com/MatterOverdrive/MatterOverdrive/blob/1.12.2/src/main/java/matteroverdrive/imc/MOIMCHandler.java).


CHANGELOG.md
MatterOverdrive - Changelog
===========================

Unreleased
----------
* do a really stupid fix because forge is stupid, closes #117 (TheCodedOne)
* Fix villager profession, closes #112 (TheCodedOne)

0.6.0.144-BETA-4 (2018-03-23 18:32:55 +0000)
--------------------------------------------
* Fix a crash, closes #105 (TheCodedOne)
* Add config for @Vyraal1 related to #102 (TheCodedOne)
* fix a name (TheCodedOne)
* Remove Google Analytics (TheCodedOne)
* Fix config location (TheCodedOne)
* Add a config to disable entities (TheCodedOne)
* Remove legacy starmap things (TheCodedOne)
* Fix colored tritanium crate lang, closes #101 (TheCodedOne)
* Remove useless util method (TheCodedOne)
* Fix fusion reactor checking the state of air, closes #89 (TheCodedOne)
* Update README.md (TheCodedOne)
* Rotate minimap 180 deg to correctly align (#99) (Shadowfacts)
* Fix #97 (TheCodedOne)
* Remove some more legacy resources (TheCodedOne)
* Rename the pattern drive slot, closes #95 (TheCodedOne)
* Move world gen initialization to pre-init (#92) (Shadowfacts)
* Fix solar panels occluding adjacent blocks (#93) (Shadowfacts)
* stop stalebot from stale-ing features (TheCodedOne)
* Remove space-related objects from MO (TheCodedOne)
* fix deobf & api jar (TheCodedOne)
* Make an effort to start cleaning up code but really its a waste of time (TheCodedOne)
* Update MOSounds enum with new sounds (TheCodedOne)
* fine tune stale (TheCodedOne)
* move planet rendering to the sky renderer (TheCodedOne)
* add stale.yml (TheCodedOne)
* Improve space handling a bit (TheCodedOne)
* improve clouds (TheCodedOne)
* Add cloud layer (TheCodedOne)
* Texture the plaent (TheCodedOne)
* fix hwyla crash (TheCodedOne)
* Remove unused items (TheCodedOne)
* Remove curse publish for now (TheCodedOne)
* [curse publish] (TheCodedOne)
* Source not sources, [curse publish] (TheCodedOne)
* add curse building [curse publish] (TheCodedOne)
* Use alternative logo icon (TheCodedOne)
* Cleanup (TheCodedOne)
* make changelog generate a full changelog (TheCodedOne)
* fix hwyla compat (TheCodedOne)
* Implement the space dimension (TheCodedOne)
* Retexture a few blocks (TheCodedOne)
* Update zh_CN.lang (#78) (ETStarLight)

0.6.0.108-BETA-3 (2018-02-28 11:03:18 +0000)
--------------------------------------------
* Compress all the things (Much smaller jar size) (TheCodedOne)
* Disable matter pipe recipes (for now) (TheCodedOne)
* Begin work on new pipe system (TheCodedOne)
* Cleanup (TheCodedOne)
* Implement a custom model loader & clean proxies a bit (TheCodedOne)
* Fix omnitool not breaking blocks, closes #44 (TheCodedOne)
* Shrink weapon texture size to improve ram usage (TheCodedOne)
* Cleanup (TheCodedOne)
* Make energy weapons not fire in spectator mode, fixes #82 (TheCodedOne)
* Fix up the mad scientist a bit (TheCodedOne)
* Improve android command (TheCodedOne)
* Make boundingbox simulate original block in pickblock (TheCodedOne)
* add coloured tritanium crates (TheCodedOne)
* Improve physbeam texture (TheCodedOne)
* add glass to the lang (TheCodedOne)
* Fix the iscollidable checks (TheCodedOne)
* Reimplement Alien dimension (TheCodedOne)
* Make the teleport biostat disable hotbar items after teleport (TheCodedOne)
* Fix the glow on the teleport biostat (TheCodedOne)
* Remove legacy code (TheCodedOne)
* Formatting (TheCodedOne)
* Disabling shift clicking inside inventories until fixed, cc #73 (TheCodedOne)
* Remove some legacy code (TheCodedOne)
* Fullbright a bunch of models (TheCodedOne)
* add IndustrialGlass (TheCodedOne)
* brighten the crate overlay texture for later usage (TheCodedOne)
* Add CTM (TheCodedOne)
* Dont add to loot table (TheCodedOne)
* Bunch of stuff, read description for actual info (TheCodedOne)
* Reimplement wireless charging, closes #79 (TheCodedOne)
* Finish the version checker update, all is working now (TheCodedOne)
* Update the version handler (TheCodedOne)
* Cleanup (TheCodedOne)
* Remove slots from starmap (TheCodedOne)
* remove energy unit config (TheCodedOne)
* Make gravitational anomaly non-collidable (TheCodedOne)
* Fix sonic shockwave ignoring pvp, closes #59 (TheCodedOne)
* Fix inventories doing.. pretty retarded shit (TheCodedOne)
* Fix teleporting into blocks that you shouldn't (TheCodedOne)
* Fix TESRs rendering when they shouldn't, closes #58 (TheCodedOne)
* Fix inscriber TESR (TheCodedOne)
* Fix weapon station not listening to the slot (TheCodedOne)
* add override (TheCodedOne)
* Fix up some null stacks (TheCodedOne)
* Whoops wrong class (TheCodedOne)
* Finish up overclock enchantment a bit (TheCodedOne)
* Fix wrench animation (TheCodedOne)
* Fix weapon reequip animation (TheCodedOne)
* Fix #71 (TheCodedOne)
* Fix some dupe bugs (TheCodedOne)
* Reimplment old inscriber recipes until CLXML is stable (TheCodedOne)
* Fix android hud rendering weirdness (TheCodedOne)
* Fix tritanium armor (TheCodedOne)
* Fix hundreds of anomalies spawning, closes #66 (TheCodedOne)
* fix serverside method (TheCodedOne)
* Fix a few issues (TheCodedOne)
* Fixed WeaponFactory checks (#65) (DBotThePony)
* Correct repository URL in the git clone command (#60) (IncognitoJam)
* Remove inscriber recipes in preparation (TheCodedOne)
* Register MO Enchantments (TheCodedOne)
* Improve some stack & state logic (TheCodedOne)

0.6.0.75-BETA-2 (2018-02-16 08:58:54 +0000)
-------------------------------------------
* Fix android hud up, closes #5 (TheCodedOne)
* General cleanup (TheCodedOne)
* Update zh_cn.lang (#55) (冷风)
* Weird (TheCodedOne)
* Don't rotate HUD if player is sleeping (#54) (DBotThePony)
* Fixed HUD is rendered while in spectator mode (#53) (DBotThePony)
* Fixed energy issues (#50) (DBotThePony)
* Begin planning out new sounds (TheCodedOne)
* Revert "Work on fixing energy items" (TheCodedOne)
* Work on fixing energy items (TheCodedOne)
* some reformatting (TheCodedOne)
* Fix the inscriber up a bit, closes #41 (TheCodedOne)
* Fix level+1 (TheCodedOne)
* Fix some more loottable name stuff (TheCodedOne)

src/main/java/matteroverdrive/Reference.java
/*
 * This file is part of Matter Overdrive
 * Copyright (c) 2015., Simeon Radivoev, All rights reserved.
 *
 * Matter Overdrive is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Matter Overdrive is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Matter Overdrive.  If not, see <http://www.gnu.org/licenses>.
 */

package matteroverdrive;

import matteroverdrive.client.data.Color;

public final class Reference {
    public static final String MOD_ID = "matteroverdrive";
    public static final String MOD_NAME = "MatterOverdrive";
    public static final String VERSION = "@VERSION@";
    public static final String VERSION_DATE = "@DATE@";
    public static final String GUI_FACTORY_CLASS = "matteroverdrive.gui.GuiConfigFactory";
    public static final String DEPENDENCIES = "required-after:clib@[1.0.2.31,)";
    public static final String CHANNEL_NAME = "mo_channel";
    public static final String CHANNEL_WEAPONS_NAME = CHANNEL_NAME + ":weapons";
    public static final String CHANNEL_GUI_NAME = CHANNEL_NAME + ":gui";
    public static final String VERSIONS_CHECK_URL = "https://api.cfwidget.com/mc-mods/minecraft/229694-matter-overdrive";
    public static final String BETA_CHECK_URL = "http://maven.k-4u.nl/matteroverdrive/MatterOverdrive/maven-metadata.xml";
    public static final String DOWNLOAD_URL = "https://minecraft.curseforge.com/projects/matter-overdrive";
    //region GUI
    public static final String PATH_GFX = MOD_ID + ":textures/";
    public static final String PATH_WORLD_TEXTURES = PATH_GFX + "world/";
    public static final String PATH_ARMOR = PATH_GFX + "armor/";
    public static final String PATH_PARTICLE = PATH_GFX + "particle/";
    public static final String PATH_GUI = PATH_GFX + "gui/";
    public static final String PATH_BLOCKS = PATH_GFX + "blocks/";
    public static final String PATH_FX = PATH_GFX + "fx/";
    public static final String PATH_SHADERS = MOD_ID + ":shaders/";
    public static final String PATH_ELEMENTS = PATH_GUI + "elements/";
    public static final String PATH_ICON = PATH_GUI + "icons/";
    public static final String PATH_ENTITIES = PATH_GFX + "entities/";
    public static final String PATH_ITEM = PATH_GFX + "items/";
    public static final String PATH_GUI_ITEM = PATH_GUI + "items/";
    public static final String PATH_MODEL = MOD_ID + ":models/";
    public static final String PATH_MODEL_BLOCKS = PATH_MODEL + "block/";
    public static final String PATH_MODEL_ITEMS = PATH_MODEL + "item/";
    public static final String PATH_SOUNDS = MOD_ID + ":sounds/";
    public static final String PATH_SOUNDS_BLOCKS = PATH_SOUNDS + "blocks/";
    public static final String PATH_INFO = MOD_ID + ":info/";
    //region GUI Textures
    public static final String TEXTURE_ARROW_PROGRESS = PATH_ELEMENTS + "progress_arrow_right.png";
    //endregion
    public static final String TEXTURE_FE_METER = PATH_ELEMENTS + "fe.png";
    //region Colors
    public static final Color COLOR_WHITE = new Color(255, 255, 255);
    public static final Color COLOR_MATTER = new Color(191, 228, 230);
    public static final Color COLOR_HOLO = new Color(169, 226, 251);
    public static final Color COLOR_YELLOW_STRIPES = new Color(254, 203, 4);
    public static final Color COLOR_HOLO_RED = new Color(230, 80, 20);
    public static final Color COLOR_HOLO_GREEN = new Color(24, 207, 0);
    public static final Color COLOR_HOLO_YELLOW = new Color(252, 223, 116);
    public static final Color COLOR_HOLO_PURPLE = new Color(116, 23, 230);
    public static final Color COLOR_GUI_NORMAL = new Color(62, 81, 84);
    public static final Color COLOR_GUI_LIGHT = new Color(100, 113, 136);
    public static final Color COLOR_GUI_LIGHTER = new Color(139, 126, 168);
    public static final Color COLOR_GUI_DARK = new Color(44, 54, 52);
    public static final Color COLOR_GUI_DARKER = new Color(34, 40, 37);
    public static final Color COLOR_GUI_ENERGY = new Color(224, 0, 0);
    //region Modules
    public static final int MODULE_BATTERY = 0;
    //endregion
    public static final int MODULE_COLOR = 1;
    public static final int MODULE_BARREL = 2;
    public static final int MODULE_SIGHTS = 3;
    public static final int MODULE_OTHER = 4;
    //region Bionic Types
    public static final int BIONIC_HEAD = 0;
    //end region
    public static final int BIONIC_ARMS = 1;
    public static final int BIONIC_LEGS = 2;
    public static final int BIONIC_CHEST = 3;
    public static final int BIONIC_OTHER = 4;
    public static final int BIONIC_BATTERY = 5;
    //region Weapon Stat
    public static final int WS_DAMAGE = 0;
    //endregion
    public static final int WS_AMMO = 1;
    public static final int WS_EFFECT = 2;
    public static final int WS_RANGE = 3;
    public static final int WS_FIRE_DAMAGE = 4;
    public static final int WS_BLOCK_DAMAGE = 5;
    public static final int WS_EXPLOSION_DAMAGE = 6;
    public static final int WS_FIRE_RATE = 7;
    public static final int WS_HEAL = 8;
    public static final int WS_MAX_HEAT = 9;
    public static final int WS_ACCURACY = 10;
    public static final int WS_SHOOT_COOLDOWN = 11;
    public static final int WS_RICOCHET = 12;
    //region Request Packet Type
    public static final int PACKET_REQUEST_CONNECTION = 0;
    //endregion
    public static final int PACKET_REQUEST_PATTERN_SEARCH = 1;
    public static final int PACKET_REQUEST_NEIGHBOR_CONNECTION = 2;
    public static final int PACKET_REQUEST_VALID_PATTERN_DESTINATION = 3;
    //region Broadcast Packet Type
    public static final int PACKET_BROADCAST_CONNECTION = 0;
    //endregion
    //region Packet Responce Type
    public static final int PACKET_RESPONCE_ERROR = -1;
    //endregion
    public static final int PACKET_RESPONCE_INVALID = 0;
    public static final int PACKET_RESPONCE_VALID = 1;
    //region machine mods
    public static final byte MODE_REDSTONE_NONE = 2;
    //endregion
    public static final byte MODE_REDSTONE_HIGH = 1;
    public static final byte MODE_REDSTONE_LOW = 0;
    //region models
    public static final String MODEL_SPHERE = PATH_MODEL_BLOCKS + "sphere.obj";
    //endregion
    public static final String MODEL_CHARGING_STATION = PATH_MODEL_BLOCKS + "charging_station.obj";
    public static final String MODEL_PATTERN_STORAGE = PATH_MODEL_BLOCKS + "pattern_storage.obj";
    public static final String MODEL_REPLICATOR = PATH_MODEL_BLOCKS + "replicator.obj";
    public static final String MODEL_TRITANIUM_CRATE = PATH_MODEL_BLOCKS + "tritanium_crate.obj";
    public static final String MODEL_INSCRIBER = PATH_MODEL_BLOCKS + "inscriber.obj";
    //region config keys
    public static final String CONFIG_KEY_REDSTONE_MODE = "redstoneMode";
    //endregion
    //region World Geb
    public static final String CHEST_GEN_ANDROID_HOUSE = "android_house";
    //endregion
    public static final String WORLD_DATA_MO_GEN_POSITIONS = "MatterOverdriveWorldGenPositions";
    //region Unicodes
    public static final String UNICODE_LEGENDARY = "\u272a";
    //endregion
    public static final String UNICODE_COMPLETED_OBJECTIVE = "\u25a0";
    public static final String UNICODE_UNCOMPLETED_OBJECTIVE = "\u25a1";
    private Reference() {
    }
    //endregion
}
